cmd/mdatagen/internal/templates/config.go.tmpl

```diff
@@ -12,7 +14,8 @@ import (
 
 // MetricConfig provides common config for a particular metric.
 type MetricConfig struct {
-	Enabled bool `mapstructure:"enabled"`
+	Enabled             bool   `mapstructure:"enabled"`
+  AggregationStrategy string `mapstructure:"aggregation_strategy"`
```

Comment:
Should the unmarshal fail if a non-valid "aggregation_strategy" is defined?

Reply:
I was also wondering where it should fail in the invalid strategy case. I was leaning to having the `MetricConfig` struct implement its own `Validate()` but on unmarshal also makes sense to me.